helix-term/src/commands.rs

```diff
@@ -1070,8 +1070,8 @@ mod cmd {
             .filter(|doc| doc.is_modified())
             .map(|doc| {
                 doc.relative_path()
-                    .and_then(|path| path.to_str())
-                    .unwrap_or("[scratch]")
+                    .map(|path| path.to_string_lossy().to_string())
```

Comment:
No need for `to_string()` here, it's already a `Cow` I think.

Reply:
Unfortunatelly:
```rust
error[E0515]: cannot return value referencing function parameter `path`
    --> helix-term/src/commands.rs:1073:33
     |
1073 |                     .map(|path| path.to_string_lossy())
     |                                 ----^^^^^^^^^^^^^^^^^^
     |                                 |
     |                                 returns a value referencing data owned by the current function
     |                                 `path` is borrowed here
```

Reply:
It is a consequence of changing the return type of `relative_path` to be owned `PathBug` not `&Path` but I'm not sure I can return a `&Path` while still folding home directory into tilde.